plugins/modules/packaging/os/yum_versionlock.py

```diff
@@ -125,18 +140,11 @@ def main():
     packages_list = []
     if state in ('present'):
         command = 'add'
-        for single_pkg in packages:
-            if not any(fnmatch(pkg.split(":", 1)[-1], single_pkg) for pkg in versionlock_packages.split()):
-                packages_list.append(single_pkg)
-        if packages_list:
-            if module.check_mode:
-                changed = True
-            else:
-                changed = yum_v.ensure_state(packages_list, command)
     elif state in ('absent'):
         command = 'delete'
+    if state in ('present', 'absent'):
         for single_pkg in packages:
-            if any(fnmatch(pkg, single_pkg) for pkg in versionlock_packages.split()):
+            if not any(match(pkg, single_pkg) for pkg in versionlock_packages.split()):
```

Comment:
This change is wrong, resp. only works for `state == present`, but **not** for `state == absent`.

Having this loop twice, the one without `not` and the other with `not` before the `any()` is needed.